android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -736,21 +731,31 @@ public int getInputType()
 												'<', '>', ',', '?', '/', ':', ';', '\'', '"', '~' };
 						}
 					});
+					customKeyListener = true;
 					break;
 				case KEYBOARD_URL:
 					Log.d(TAG, "Setting keyboard type URL-3", Log.DEBUG_MODE);
 					tv.setImeOptions(EditorInfo.IME_ACTION_GO);
 					textTypeAndClass |= InputType.TYPE_TEXT_VARIATION_URI;
 					break;
 				case KEYBOARD_DECIMAL_PAD:
-					textTypeAndClass = (InputType.TYPE_NUMBER_FLAG_DECIMAL | InputType.TYPE_NUMBER_FLAG_SIGNED);
+					tv.setKeyListener(new DigitsKeyListener(Locale.getDefault(), true, true) {
```

Comment:
The `new DigitskeyListener()` with the `Locale` argument is API Level 26 only. So, this will likely crash on an OS older than Android 8.0. But we definitely need to call this constructor on Android 8.0 and higher since it's the only official way to get it to support a localized decimal point as can be seen in Google's code below...
[DigitsKeyListener.java#L202](https://github.com/aosp-mirror/platform_frameworks_base/blob/d87727f7d9350fd94831c5c5e138f9400e1823fd/core/java/android/text/method/DigitsKeyListener.java#L202)

Do you have any ideas on how to get this working on older Android OS versions?

Since Google's source code is Apache licensed, we can copy their current code into our `TiDigitsKeyListener` and use that for the older OS versions only. Effectively, we'd be backporting it like how Google backports it via their support libraries. Ideally, we should avoid this unless we have no other choice.